### PR TITLE
[RDBMS] Add List-SKUS Command, Table Transformers, Local Context for Postgres, MySQL, Mariadb Single Server

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**  

1. Added list-skus commands for single server Postgres and MySQL
2. Added table transformers for list-skus and server list commands
3. Added local-context support for single server

**Testing Guide**  

1. List-skus command: run with the below commands
- az postgres server list-skus -l eastus -o table
- az mysql server list-skus -l eastus -o table
- az mariadb server list-skus -l eastus -o table

2. server list command: run with the below commands
- az postgres server list -g <group_name> -n <server_name> -o table
- az mysql server list -g <group_name> -n <server_name> -o table
- az mariadb server list -g <group_name> -n <server_name> -o table

3. local context command: 
az config param-persist on
- az postgres -h and run commands to check if server name argument is included in the local context
e.g.) az postgres server create -g <group_name> -n <server_name> -l <location> => az postgres server show

**History Notes**  

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/_transformers.py
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/_transformers.py:61]: Nit : This is exactly same as table_transform_output_list_servers except haState and availablityZone. So try and see either

 1. If you can pass a parameter to the function and based on the parameter value( server/flexible server) , determine whether or not to add those two fields, or
 2. Have a common function and remove the repeated code there and make both these functions call the common function wherever applicable and add the rest of the exclusive fields in the respective caller.

File: src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:281]: Do we not have this for MariaDB ?

File: src/azure-cli/azure/cli/command_modules/rdbms/_help.py
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/_help.py:1224]: If you add for this command for MariaDb, update this as well

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:129]: I hope you have validated/ tested these changes

File: src/azure-cli/azure/cli/command_modules/rdbms/commands.py
Comment (arde0708) [inline src/azure-cli/azure/cli/command_modules/rdbms/commands.py:227]: Required for MariaDB if the API exists

File: src/azure-cli/azure/cli/command_modules/rdbms/_transformers.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_transformers.py:61]: Resolved. Thanks!!

File: src/azure-cli/azure/cli/command_modules/rdbms/_help.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_help.py:1224]: Same!

File: src/azure-cli/azure/cli/command_modules/rdbms/_help.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_help.py:1246]: Lists->List

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:43]: @arrownj could you help with local context part?

File: src/azure-cli/azure/cli/command_modules/rdbms/commands.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: please add tests for all these new commands.

Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: If there is limited api version supported, please specify min_api also.

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: Tests including these new commands are added in this pull request https://github.com/Azure/azure-cli/pull/15363

File: src/azure-cli/azure/cli/command_modules/rdbms/_help.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_help.py:1246]: will do

File: src/azure-cli/azure/cli/command_modules/rdbms/commands.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: This did exist in APIs already but we just added this to CLI just now

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/commands.py:503]: Will do

**Review by arrownj — APPROVED** [review]
Looks good for `local context` part.

**Review by Juliehzl — APPROVED** [review]
LGTM for command design

**Review by haroldrandom — APPROVED** [review]

**Review by haroldrandom — APPROVED** [review]
will merge once @arde0708 approve

**Review by arde0708 — APPROVED** [review]

**Comment by arde0708** [comment]
> will merge once @arde0708 approve

Done

**Pull request merged**
